nilearn/_utils/tests/test_param_validation.py

```diff
@@ -101,3 +105,45 @@ def test_feature_screening(affine_eye):
                     ),
                     BaseEstimator,
                 )
+
+
+@pytest.mark.xfail
+@pytest.mark.parametrize("screening_percentile", [1, 20, 90])
+def test_screening_adjustment(affine_eye, screening_percentile):
+
+    fsaverage = datasets.fetch_surf_fsaverage("fsaverage5")
+
+    mask_img_data = np.zeros((182, 218, 182))
+    mask_img_data[30:-30, 30:-30, 30:-30] = 1
+    mask_img = Nifti1Image(mask_img_data, affine=affine_eye)
+    mask_surf = vol_to_surf(
+        img=mask_img,
+        surf_mesh=fsaverage["pial_left"],
+        inner_mesh=fsaverage["white_left"],
+    )
+
+    img_data = np.random.randn(182, 218, 182)
+    img = Nifti1Image(img_data, affine=affine_eye)
+    img_surf = vol_to_surf(
+        img=img,
+        surf_mesh=fsaverage["pial_left"],
+        inner_mesh=fsaverage["white_left"],
+    )
+
+    mask_n_vertices = get_mask_volume(mask_surf)
+    mesh_n_vertices = fsaverage.mesh.n_vertices
+
+    mask_to_mesh_ratio = (mask_n_vertices / mesh_n_vertices) * 100
+
+    adjusted = adjust_screening_percentile(
+        screening_percentile, mask_to_mesh_ratio, mesh_n_vertices
+    )
+
+    if screening_percentile == 100:
+        adjusted = 100
+    if mask_to_mesh_ratio < screening_percentile:
+        adjusted = 100
+    elif mask_to_mesh_ratio > screening_percentile:
+        adjusted = screening_percentile * (mesh_n_vertices / mask_n_vertices)
+    elif mask_to_mesh_ratio == screening_percentile:
+        adjusted = screening_percentile
```

Comment:
I wanted to test the screening percentile adjustment somewhat like this. However, I am struggling to create an appropriate sample mask and image.

I saw that there are some fixtures defined in `nilearn/experimental/conftest.py` but I couldn't call them here (even after importing them in `nilearn/conftest.py`).

Any suggestions @Remi-Gau ?

Reply:
Updated the code with comments:
```python
def test_screening_adjustment(affine_eye, screening_percentile):

    # get a mesh
    fsaverage = datasets.fetch_surf_fsaverage("fsaverage5")
    # create sample mask image
    mask_img_data = np.zeros((182, 218, 182))
    mask_img_data[30:-30, 30:-30, 30:-30] = 1
    mask_img = Nifti1Image(mask_img_data, affine=affine_eye)
    mask_surf = vol_to_surf(
        img=mask_img,
        surf_mesh=fsaverage["pial_left"],
        inner_mesh=fsaverage["white_left"],
    )
    # create sample image
    img_data = np.random.randn(182, 218, 182)
    img = Nifti1Image(img_data, affine=affine_eye)
    img_surf = vol_to_surf(
        img=img,
        surf_mesh=fsaverage["pial_left"],
        inner_mesh=fsaverage["white_left"],
    )
    # get number of vertices in mask and mesh
    mask_n_vertices = get_mask_volume(mask_surf)
    mesh_n_vertices = fsaverage.mesh.n_vertices
    # get mask to mesh ratio
    mask_to_mesh_ratio = (mask_n_vertices / mesh_n_vertices) * 100
    # adjust screening percentile
    adjusted = adjust_screening_percentile(
        screening_percentile, mask_to_mesh_ratio, mesh_n_vertices
    )
    if screening_percentile == 100:
        assert adjusted == 100
    # if mask is smaller than given percentile, full mask is used
    if mask_to_mesh_ratio < screening_percentile:
        assert adjusted == 100
    # if mask is larger than given percentile, the percentile is adjusted to
    # the ratio of mesh to mask
    elif mask_to_mesh_ratio > screening_percentile:
        assert adjusted == screening_percentile * (
            mesh_n_vertices / mask_n_vertices
        )
    # if mask is equal to given percentile, the adjusted percentile is the
    # same as the given percentile
    elif mask_to_mesh_ratio == screening_percentile:
        assert adjusted == screening_percentile
```

Reply:
This could be deferred to a separate issue for later. Let me know what you think!

Reply:
Yeah I am tempted to put this in a separate issue / PR.
Would rather have smaller PRs.

Reply:
Ok I will open a separate issue then